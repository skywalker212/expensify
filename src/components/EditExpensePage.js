import React from 'react';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses';
import {connect} from 'react-redux';

const EditExpensePage = (props) => {
    return (
        <div>
            <ExpenseForm expense={props.expense} onSubmit={(expense)=>{
               props.dispatch(editExpense(
                   props.expense.id,
                   {...expense}
               ));
               props.history.push('/');
            }}/>
            <button onClick={() => {
                props.dispatch(removeExpense({id: props.expense.id}));
                props.history.push('/');
            }}>Remove</button>
        </div>
    );
};

const mapStateToProps = (state,props) => ({
    expense: state.expenses.find( item => item.id===props.match.params.id )
});

export default connect(mapStateToProps)(EditExpensePage);