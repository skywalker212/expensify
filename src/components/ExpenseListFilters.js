import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByDate,sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilters extends React.Component{

    state = {
        calendarFocused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (calendarFocused) => {
        this.setState(()=>({calendarFocused}));
    }

    onTextChange = (event) => {
        this.props.setTextFilter(event.target.value);
    };

    onSortchange = (e) => {
        e.target.value==='date'?this.props.sortByDate():this.props.sortByAmount();
    };

    render() {
        return (
        <div>
            <input type="text" onChange={this.onTextChange} value={this.props.filters.text}/>
            <select value={this.props.filters.sortBy} onChange={this.onSortchange}>
                <option value='date'>Date</option>
                <option value='amount'>Amount</option>
            </select>
            <DateRangePicker
             startDate={this.props.filters.startDate} 
             endDate={this.props.filters.endDate} 
             onDatesChange={this.onDatesChange}
             focusedInput={this.state.calendarFocused}
             onFocusChange={this.onFocusChange}
             startDateId="START_DATE_ID"
             endDateId="END_DATE_ID"
             numberOfMonths={1}
             isOutsideRange={()=>false}
             showClearDates={true}
             />
        </div>
        );
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);