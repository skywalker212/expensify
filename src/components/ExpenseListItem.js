import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id,amount,description,note,createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>Note: {note}</p>
        <p>{numeral(amount/100).format('$0,0.00')}
            - 
            {moment(createdAt).format('MMMM Do, YYYY')}</p>
    </div>
); 

export default ExpenseListItem;