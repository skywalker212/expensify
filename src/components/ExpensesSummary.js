import React from 'react';
import expensesSelector from '../selectors/expenses';
import selectTotalExpenses from '../selectors/expenses-total';
import {connect} from 'react-redux';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component{
    render(){
        return (
            <h2>
                Viewing {this.props.expenseCount} expense{this.props.expenseCount>1?'s':''} totaling {numeral(this.props.totalExpenses/100).format('$0,0.00')}
            </h2>
        );
    }
}

const mapStateToProps = (state) => {
    const expenses = expensesSelector(state.expenses,state.filters);
    return {
        expenseCount: expenses?expenses.length:0,
        totalExpenses: selectTotalExpenses(expenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);