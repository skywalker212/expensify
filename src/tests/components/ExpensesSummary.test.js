import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import {shallow} from 'enzyme';

test('should render expense summary correctly',()=>{
    const wrapper = shallow(<ExpensesSummary/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses summary for single expense correctly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} totalExpenses={195}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses summary for multiple expenses correctly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={3} totalExpenses={114195}/>);
    expect(wrapper).toMatchSnapshot();
});