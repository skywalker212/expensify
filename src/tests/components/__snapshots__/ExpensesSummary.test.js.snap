// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary correctly 1`] = `
<h2>
  Viewing 
   expense
   totaling 
  $0.00
</h2>
`;

exports[`should render expenses summary for multiple expenses correctly 1`] = `
<h2>
  Viewing 
  3
   expense
  s
   totaling 
  $1,141.95
</h2>
`;

exports[`should render expenses summary for single expense correctly 1`] = `
<h2>
  Viewing 
  1
   expense
   totaling 
  $1.95
</h2>
`;
